source/browser.ts

```diff
@@ -416,111 +416,109 @@ function closePreferences(): void {
-// 			const canvas = document.createElement('canvas');
-// 			const padding = {
-// 				top: 3,
-// 				right: 0,
-// 				bottom: 3,
-// 				left: 0
-// 			};
-
-// 			canvas.width = size + padding.left + padding.right;
-// 			canvas.height = size + padding.top + padding.bottom;
-
-// 			const ctx = canvas.getContext('2d')!;
-// 			ctx.save();
-// 			ctx.beginPath();
-// 			ctx.arc(size / 2 + padding.left, size / 2 + padding.top, size / 2, 0, Math.PI * 2, true);
-// 			ctx.closePath();
-// 			ctx.clip();
-// 			ctx.drawImage(img, padding.left, padding.top, size, size);
-// 			ctx.restore();
-
-// 			resolve(canvas);
-// 		});
-
-// 		img.src = url;
-// 	});
-// }
+async function urlToCanvas(url: string, size: number): Promise<HTMLCanvasElement> {
+	return new Promise(resolve => {
+		const img = new Image();
+		img.crossOrigin = 'anonymous';
+		img.addEventListener('load', () => {
+			const canvas = document.createElement('canvas');
+			const padding = {
+				top: 3,
+				right: 0,
+				bottom: 3,
+				left: 0
+			};
+
+			canvas.width = size + padding.left + padding.right;
+			canvas.height = size + padding.top + padding.bottom;
+
+			const ctx = canvas.getContext('2d')!;
+			ctx.save();
+			ctx.beginPath();
+			ctx.arc(size / 2 + padding.left, size / 2 + padding.top, size / 2, 0, Math.PI * 2, true);
+			ctx.closePath();
+			ctx.clip();
+			ctx.drawImage(img, padding.left, padding.top, size, size);
+			ctx.restore();
+
+			resolve(canvas);
+		});
+
+		img.src = url;
+	});
+}
 
 // Return data url for user avatar
-// async function getDataUrlFromImg(img: HTMLImageElement, unread: boolean): Promise<string> {
-// 	// eslint-disable-next-line no-async-promise-executor
-// 	return new Promise(async resolve => {
-// 		if (unread) {
-// 			const dataUnreadUrl = img.getAttribute('dataUnreadUrl');
-
-// 			if (dataUnreadUrl) {
-// 				return resolve(dataUnreadUrl);
-// 			}
-// 		} else {
-// 			const dataUrl = img.getAttribute('dataUrl');
-
-// 			if (dataUrl) {
-// 				return resolve(dataUrl);
-// 			}
-// 		}
-
-// 		const canvas = await urlToCanvas(img.src, 30);
-// 		const ctx = canvas.getContext('2d')!;
-// 		const dataUrl = canvas.toDataURL();
-// 		img.setAttribute('dataUrl', dataUrl);
-
-// 		if (!unread) {
-// 			return resolve(dataUrl);
-// 		}
-
-// 		const markerSize = 8;
-// 		ctx.fillStyle = '#f42020';
-// 		ctx.beginPath();
-// 		ctx.ellipse(canvas.width - markerSize, markerSize, markerSize, markerSize, 0, 0, 2 * Math.PI);
-// 		ctx.fill();
-// 		const dataUnreadUrl = canvas.toDataURL();
-// 		img.setAttribute('dataUnreadUrl', dataUnreadUrl);
-
-// 		resolve(dataUnreadUrl);
-// 	});
-// }
+async function getDataUrlFromImg(img: HTMLImageElement, unread: boolean): Promise<string> {
+	// eslint-disable-next-line no-async-promise-executor
+	return new Promise(async resolve => {
+		if (unread) {
+			const dataUnreadUrl = img.getAttribute('dataUnreadUrl');
+
+			if (dataUnreadUrl) {
+				return resolve(dataUnreadUrl);
```

Comment:
This creates misleading code as it makes it look like `resolve` actually returns something.
```suggestion
				resolve(dataUnreadUrl);
				return;
```

Reply:
I actually just reverted this piece of code from the previous version, but sure, I can make this change!

Reply:
Done!